src/renderers/dom/shared/DOMProperty.js

```diff
@@ -13,6 +13,21 @@
 
 var invariant = require('fbjs/lib/invariant');
 
+var RESERVED_PROPS = {
+  children: true,
+  dangerouslySetInnerHTML: true,
+  key: true,
```

Comment:
Good question. This goes back a really long time, and I can't remember. I'll just remove them.